pkg/webhook/gateway/v1beta1/gateway_validation.go

```diff
@@ -0,0 +1,145 @@
+func (webhook *GatewayHandler) ValidateDelete(_ context.Context, obj runtime.Object) error {
+	gw, ok := obj.(*v1beta1.Gateway)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a Gateway but got a %T", obj))
+	}
+	return validate(gw)
+}
+
+func validate(g *v1beta1.Gateway) error {
+	var errList field.ErrorList
+
+	if g.Spec.ExposeType != "" {
+		if g.Spec.ExposeType != v1beta1.ExposeTypeLoadBalancer && g.Spec.ExposeType != v1beta1.ExposeTypePublicIP {
+			fldPath := field.NewPath("spec").Child("exposeType")
+			errList = append(errList, field.Invalid(fldPath, g.Spec.ExposeType, "the 'exposeType' field  is irregularity"))
+		}
+		if g.Spec.ExposeType == v1beta1.ExposeTypeLoadBalancer || g.Spec.ExposeType == v1beta1.ExposeTypePublicIP {
+			for i, ep := range g.Spec.Endpoints {
+				if ep.UnderNAT {
+					fldPath := field.NewPath("spec").Child(fmt.Sprintf("endpoints[%d]", i)).Child("underNAT")
+					errList = append(errList, field.Invalid(fldPath, ep.UnderNAT, fmt.Sprintf("the 'underNAT' field for exposed gateway %s/%s must be false", g.Namespace, g.Name)))
+				}
+			}
+		}
+	}
+
+	if len(g.Spec.Endpoints) != 0 {
+		underNAT := g.Spec.Endpoints[0].UnderNAT
+		for i, ep := range g.Spec.Endpoints {
+			if ep.UnderNAT != underNAT {
+				fldPath := field.NewPath("spec").Child(fmt.Sprintf("endpoints[%d]", i)).Child("underNAT")
+				errList = append(errList, field.Invalid(fldPath, ep.UnderNAT, "the 'underNAT' field in endpoints must be the same"))
+			}
+			if ep.PublicIP != "" {
+				if err := validateIP(ep.PublicIP); err != nil {
+					fldPath := field.NewPath("spec").Child(fmt.Sprintf("endpoints[%d]", i)).Child("publicIP")
+					errList = append(errList, field.Invalid(fldPath, ep.PublicIP, "the 'publicIP' field must be a validate IP address"))
+				}
+			}
+			if ep.Type == "" {
+				fldPath := field.NewPath("spec").Child(fmt.Sprintf("endpoints[%d]", i)).Child("type")
+				errList = append(errList, field.Invalid(fldPath, ep.Type, "the 'type' field must not be empty"))
+			} else {
+				if ep.Type != v1beta1.Tunnel && ep.Type != v1beta1.Proxy {
+					fldPath := field.NewPath("spec").Child(fmt.Sprintf("endpoints[%d]", i)).Child("type")
+					errList = append(errList, field.Invalid(fldPath, ep.Type, fmt.Sprintf("the 'type' field must be set %s or %s ", v1beta1.Tunnel, v1beta1.Proxy)))
+				}
+			}
+			if ep.Type == v1beta1.Tunnel && ep.Port != v1beta1.TunnelPort {
+				fldPath := field.NewPath("spec").Child(fmt.Sprintf("endpoints[%d]", i)).Child("port")
+				errList = append(errList, field.Invalid(fldPath, ep.Port, fmt.Sprintf("the 'port' field must be set %d for type %s ", v1beta1.TunnelPort, v1beta1.Tunnel)))
+			}
+			if ep.Type == v1beta1.Proxy && ep.Port != v1beta1.ProxyPort {
+				fldPath := field.NewPath("spec").Child(fmt.Sprintf("endpoints[%d]", i)).Child("port")
+				errList = append(errList, field.Invalid(fldPath, ep.Port, fmt.Sprintf("the 'port' field must be set %d for type %s ", v1beta1.TunnelPort, v1beta1.Proxy)))
+			}
```

Comment:
In high availability mode, i think that there are multiple ports will be configured in Endpoints, maybe we need to set a port range(like 10262~10269) for proxy port.